.github/workflows/ci.yml

```diff
@@ -77,40 +79,28 @@ jobs:
     - name: Checkout code
       uses: actions/checkout@v4
       with:
-        fetch-depth: 0
-
-    - name: Set up Python
-      uses: actions/setup-python@v5
-      with:
-        python-version: ${{ matrix.python }}
-        cache: pip
+        persist-credentials: false
 
     - name: Install Graphviz for doc builds
-      if: startsWith(matrix.name, 'Documentation')
+      if: startsWith(matrix.nox_session, 'docs')
       uses: ts-graphviz/setup-graphviz@v2
 
-    - name: Install pandoc for doc builds (requires Ubuntu)
-      if: startsWith(matrix.name, 'Documentation')
+    - name: Install pandoc for doc builds
+      # This step requires Ubuntu
+      if: startsWith(matrix.nox_session, 'docs')
       run: |
         sudo apt update
         sudo apt-get install -y pandoc
```

Comment:
It would have been nice to make this workflow independent of OS, but I haven't found a reliable cross-platform way of installing `pandoc`.